ql/src/java/org/apache/hadoop/hive/ql/exec/Operator.java

```diff
@@ -1543,4 +1544,11 @@ public final boolean logicalEqualsTree(Operator<?> o) {
     }
     return true;
   }
+
+  public void replaceTabAlias(String oldAlias, String newAlias) {
```

Comment:
I am wondering whether `replaceTabAlias` should be in the operator conf (`OperatorDesc`). There is an `AbstractOperatorDesc` which could hold the default logic. Then it could be exposed here by simply calling `conf.replaceTabAlias` (single implementation for all operators).
The advantage that I see is that you would force any custom operator that does not inherit `AbstractOperatorDesc` to implement this method (I know we do not introduce such operators that much -or at all- but it seems to me it may make sense from the separation of concepts perspective).

Reply:
yes; it would make sense to do that - I've opened HIVE-24382